src/decorators/providers/component.js

```diff
@@ -181,3 +172,23 @@ Module.addProvider(TYPE, (target, name, injects, ngModule) => {
 	// Finally add the component to the raw module
 	ngModule.directive(name, () => ddo);
 });
+
+function checkComponentConfig(ddo) {
```

Comment:
It would be great if we could get the errors thrown by this functional to point to the class where the incorrect decorators were used.

```js
@Component({ selector: '[my-component]' })
class MyComponent{ ... }
```

would result in: 
`@Component selectors can only be elements. Perhaps you meant to use @Directive? On 'MyComponent'`